lib/Slack_web_api.js

```diff
@@ -1,314 +1,180 @@
+        var method = groups.pop();
+        var currentGroup = slack_api;
+        var nextGroupName;
+
+        for (var i = 0; i < groups.length; i++) {
+            nextGroupName = groups[i];
+            if (!currentGroup[nextGroupName]) {
+                currentGroup[nextGroupName] = {};
             }
-        },
-        reactions: {
-            add: function(options, cb) {
-                slack_api.callAPI('reactions.add', options, cb);
-            },
-            get: function(options, cb) {
-                slack_api.callAPI('reactions.get', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('reactions.list', options, cb);
-            },
-            remove: function(options, cb) {
-                slack_api.callAPI('reactions.remove', options, cb);
-            },
-        },
-        rtm: {
-            start: function(options, cb) {
-                slack_api.callAPI('rtm.start', options, cb);
-            },
-        },
-        search: {
-            all: function(options, cb) {
-                slack_api.callAPI('search.all', options, cb);
-            },
-            files: function(options, cb) {
-                slack_api.callAPI('search.files', options, cb);
-            },
-            messages: function(options, cb) {
-                slack_api.callAPI('search.messages', options, cb);
-            },
-        },
-        stars: {
-            list: function(options, cb) {
-                slack_api.callAPI('stars.list', options, cb);
-            },
-        },
-        team: {
-            accessLogs: function(options, cb) {
-                slack_api.callAPI('team.accessLogs', options, cb);
-            },
-            info: function(options, cb) {
-                slack_api.callAPI('team.info', options, cb);
-            },
-        },
-        users: {
-            getPresence: function(options, cb) {
-                slack_api.callAPI('users.getPresence', options, cb);
-            },
-            info: function(options, cb) {
-                slack_api.callAPI('users.info', options, cb);
-            },
-            list: function(options, cb) {
-                slack_api.callAPI('users.list', options, cb);
-            },
-            setActive: function(options, cb) {
-                slack_api.callAPI('users.setActive', options, cb);
-            },
-            setPresence: function(options, cb) {
-                slack_api.callAPI('users.setPresence', options, cb);
-            },
+            currentGroup = currentGroup[nextGroupName];
         }
+
+        currentGroup[method] = function(options, cb) {
+            slack_api.callAPI(slackMethod, options, cb);
+        };
+
+    });
+
+    // overwrite default behavior
+    slack_api.chat.postMessage = function(options, cb) {
+        if (options.attachments && typeof(options.attachments) != 'string') {
+            options.attachments = JSON.stringify(options.attachments);
```

Comment:
Alrighty; what should be the behavior in case JSON parsing fails? Seems that the method should fail, maybe I can throw a more friendly error message? On the other hand, it does seems more informative to let the raw error bubble out, since that will be actionable by the developer.